### ISSUE TITLE
configure --with-system-standard-packages={force,yes,no}

### ISSUE THREAD

**Issue by mkoeppe** [sagemath/sage#33345]
<div id="comment:0"></div>

This new option sets a default for `--with-system-...` for all standard packages that have this configure option (except for `gcc`, which has its own special default).

The default is unchanged and corresponds to the option value `yes`.

To build all standard packages (except gcc) from SPKG, use `configure --with-system-standard-packages=no`. This is a quick way to build a vanilla Sage, much shorter than using `configure --with-system-openblas=no --with-system-gfan=no --with-system-gmp=no ...........`

To require all standard packages to be present in the system, use `configure --with-system-standard-packages=force`. (This is for #30845: For example, `EXTRA_CONFIGURE_ARGS="--prefix=/opt/conda --with-system-standard-packages=force" tox -e docker-conda-forge-standard -- config.status` will test that conda packages for all standard packages (with `spkg-configure.m4`) are usable.)

The defaults set with the new option can be overridden by `--with-system-...` options for individual packages. The order in which these options appear on the command line does not matter. For example:

```
  ./configure --with-system-palp=no --with-system-standard-packages=force --with-system-pari=yes 
```

We also simplify `tox.ini` and `build/bin/write-dockerfile.sh` by removing code that generated long lists of `--with-system-...` options. It is no longer needed.

Depends on #29114

CC:  @dimpase @jhpalmieri @tobiasdiez @seblabbe @slel

Component: **build: configure**

Author: **Matthias Koeppe**

Branch/Commit: **[u/mkoeppe/configure___with_system_standard_packages__force_yes_no_](https://github.com/sagemath/sagetrac-mirror/tree/u/mkoeppe/configure___with_system_standard_packages__force_yes_no_) @ [`106c7ac`](https://github.com/sagemath/sagetrac-mirror/commit/106c7ac44e0c3a4b5c1a85c58cca417c68cb4402)**

_Issue created by migration from https://trac.sagemath.org/ticket/33345_

**Comment by mkoeppe** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -1,6 +1,15 @@
 This new option sets a default for all standard packages.
+
+The default is unchanged and corresponds to the option value `yes`.
 
 To build all standard packages (except gcc) from SPKG, use `configure --with-system-standard-packages=no`.
 
 To require all standard packages to be present in the system, use `configure --with-system-standard-packages=force`. (This is for #30845.)
 
+The defaults set with the new option can be overridden by `--with-system-...` options for individual packages. The order in which these options appear on the command line does not matter. For example
+
+```
+  ./configure --with-system-palp=no --with-system-standard-packages=force --with-system-pari=yes 
+```
+
+
``````

**Comment by mkoeppe** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -4,7 +4,7 @@
 
 To build all standard packages (except gcc) from SPKG, use `configure --with-system-standard-packages=no`.
 
-To require all standard packages to be present in the system, use `configure --with-system-standard-packages=force`. (This is for #30845.)
+To require all standard packages to be present in the system, use `configure --with-system-standard-packages=force`. (This is for #30845: For example, `EXTRA_CONFIGURE_ARGS="--prefix=/opt/conda --with-system-standard-packages=force" tox -e docker-conda-forge-standard -- config.status` will test that conda packages are usable.)
 
 The defaults set with the new option can be overridden by `--with-system-...` options for individual packages. The order in which these options appear on the command line does not matter. For example
 
``````

**Comment by mkoeppe** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -4,9 +4,9 @@
 
 To build all standard packages (except gcc) from SPKG, use `configure --with-system-standard-packages=no`.
 
-To require all standard packages to be present in the system, use `configure --with-system-standard-packages=force`. (This is for #30845: For example, `EXTRA_CONFIGURE_ARGS="--prefix=/opt/conda --with-system-standard-packages=force" tox -e docker-conda-forge-standard -- config.status` will test that conda packages are usable.)
+To require all standard packages to be present in the system, use `configure --with-system-standard-packages=force`. (This is for #30845: For example, `EXTRA_CONFIGURE_ARGS="--prefix=/opt/conda --with-system-standard-packages=force" tox -e docker-conda-forge-standard -- config.status` will test that conda packages for all standard packages (with spkg-`configure.m4`) are usable.)
 
-The defaults set with the new option can be overridden by `--with-system-...` options for individual packages. The order in which these options appear on the command line does not matter. For example
+The defaults set with the new option can be overridden by `--with-system-...` options for individual packages. The order in which these options appear on the command line does not matter. For example:
 
 ```
   ./configure --with-system-palp=no --with-system-standard-packages=force --with-system-pari=yes 
``````

**Comment by mkoeppe** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -2,7 +2,7 @@
 
 The default is unchanged and corresponds to the option value `yes`.
 
-To build all standard packages (except gcc) from SPKG, use `configure --with-system-standard-packages=no`.
+To build all standard packages (except gcc) from SPKG, use `configure --with-system-standard-packages=no`. This is a quick way to build a vanilla Sage, much shorter than using `configure --with-system-openblas=no --with-system-gfan=no --with-system-gmp=no ...........`
 
 To require all standard packages to be present in the system, use `configure --with-system-standard-packages=force`. (This is for #30845: For example, `EXTRA_CONFIGURE_ARGS="--prefix=/opt/conda --with-system-standard-packages=force" tox -e docker-conda-forge-standard -- config.status` will test that conda packages for all standard packages (with spkg-`configure.m4`) are usable.)
 
``````

**Comment by mkoeppe** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -1,4 +1,4 @@
-This new option sets a default for all standard packages.
+This new option sets a default for `--with-system-...` for all standard packages.
 
 The default is unchanged and corresponds to the option value `yes`.
 
``````

**Comment by mkoeppe** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -4,7 +4,7 @@
 
 To build all standard packages (except gcc) from SPKG, use `configure --with-system-standard-packages=no`. This is a quick way to build a vanilla Sage, much shorter than using `configure --with-system-openblas=no --with-system-gfan=no --with-system-gmp=no ...........`
 
-To require all standard packages to be present in the system, use `configure --with-system-standard-packages=force`. (This is for #30845: For example, `EXTRA_CONFIGURE_ARGS="--prefix=/opt/conda --with-system-standard-packages=force" tox -e docker-conda-forge-standard -- config.status` will test that conda packages for all standard packages (with spkg-`configure.m4`) are usable.)
+To require all standard packages to be present in the system, use `configure --with-system-standard-packages=force`. (This is for #30845: For example, `EXTRA_CONFIGURE_ARGS="--prefix=/opt/conda --with-system-standard-packages=force" tox -e docker-conda-forge-standard -- config.status` will test that conda packages for all standard packages (with `spkg-configure.m4`) are usable.)
 
 The defaults set with the new option can be overridden by `--with-system-...` options for individual packages. The order in which these options appear on the command line does not matter. For example:
 
``````

**Comment by mkoeppe** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -12,4 +12,4 @@
   ./configure --with-system-palp=no --with-system-standard-packages=force --with-system-pari=yes 
 ```
 
-
+We also simplify `tox.ini` and `build/bin/write-dockerfile.sh` by removing code that generated long lists of `--with-system-...` options. It is no longer needed.
``````

**Comment by mkoeppe** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -1,4 +1,4 @@
-This new option sets a default for `--with-system-...` for all standard packages.
+This new option sets a default for `--with-system-...` for all standard packages that have this configure option.
 
 The default is unchanged and corresponds to the option value `yes`.
 
``````

**Comment by mkoeppe** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -1,4 +1,4 @@
-This new option sets a default for `--with-system-...` for all standard packages that have this configure option.
+This new option sets a default for `--with-system-...` for all standard packages that have this configure option (except for `gcc`, which has its own special default).
 
 The default is unchanged and corresponds to the option value `yes`.
 
``````
